.github/workflows/docs.yml

```diff
@@ -51,27 +83,32 @@ jobs:
       cancel-in-progress: false
     runs-on: ubuntu-latest
     needs: docs-build
+    environment: github-pages #  Requires approval before execution
```

Comment:
I just noticed the entire idea / concept should be documented somewhere. (separate PR, so we can get this one finally merged)